### PR TITLE
Learning how to Create a PR

### PR DESCRIPTION
**Pull request by evan-m-jackson**

### CODE REVIEW

File: lib/player_choice.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class PlayerChoice
+    def is_user_input_valid(choice)
+        if choice !~ /\D/ && choice.length > 0
+          num = choice.to_i
+          if num <= 9 && num >= 1
```
Comment (jchung722) [inline lib/player_choice.rb]: You could also do a `return num <= 9 && num >= 1` instead of the conditionals

File: spec/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require 'rspec'
+require './lib/board'
+
+describe 'Board' do
+    before(:each) do
+      @board = Board.new
+    end
+  
+    it 'Has String array (aka the board)' do
+    expect(@board.board).to eq(%w[1 2 3 4 5 6 7 8 9])
```
Comment (jchung722) [inline spec/board_spec.rb]: should have indent here?

File: spec/game_over_spec.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        @board = %w[X O O 4 X 6 7 8 X]
+        expect(@game_over.which_player_wins(@board)).to eq 'X'
+    end
+  
+    it 'Player O wins diagonally the opposite way' do
+        @board = %w[X X O X O 6 O 8 9]
+        expect(@game_over.which_player_wins(@board)).to eq 'O'
+    end
+  
+    it 'returns true if Player X wins' do
```
Comment (jchung722) [inline spec/game_over_spec.rb:62]: A good way to organize specs in a class with multiple methods is separating them into describe blocks. For example:
```
describe '.does_a_player_win' do
        it 'returns true if Player X wins' do
            @board = %w[X O O 4 X 6 7 8 X]
            expect(@game_over.does_a_player_win(@board)).to be true
        end
    
        it 'returns true if Player O wins' do
            @board = %w[X X O X O 6 O 8 9]
            expect(@game_over.does_a_player_win(@board)).to be true
        end
    
        it 'returns false if there is no winner' do
            @board = %w[X O X O X O 7 8 9]
            expect(@game_over.does_a_player_win(@board)).to be false
        end
end
```

For reference, a `.` in front of a method refers to an instance method, and a `#` refers to a class method in ruby.

File: spec/player_choice_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    
+    it 'When the user input is valid' do
+      expect(@choice.is_user_input_valid('5')).to be true
+    end
+  
+    it 'When the user inputs an incorrect number' do
+      expect(@choice.is_user_input_valid('0')).to be false
+    end
+  
+    it 'When the user inputs an incorrect number' do
```
Comment (jchung722) [inline spec/player_choice_spec.rb]: Could be more specific in naming these tests, such as `returns false when user inputs a string`/ `returns false when user input is empty`

File: spec/ticTacToe_spec.rb [hunk 5]
```diff
@@ -11,56 +12,43 @@
...
-end
+  it 'Mark the board in the middle with an X' do
+    @board.mark_board('5', 'X')
+    expect(@board.board[4]).to eq('X')
+  end
 
-describe 'Game' do
+end
 
+describe 'PlayerChoice' do
```
Comment (jchung722) [inline spec/ticTacToe_spec.rb]: You forgot to move `PlayerChoice` specs out of `ticTacToe_spec.rb`

File: ticTacToe.rb [hunk 6]
```diff
@@ -49,31 +32,14 @@ def get_player_input
...
   end
 
-  def mark_board(choice)
-    @board[choice.to_i - 1] = get_marker if is_user_input_valid(choice) && is_space_free(choice)
-  end
-
   def move_confirmation(choice)
     if !is_user_input_valid(choice)
-      puts "Sorry that is not a valid move. Please select an integer between 0 and 8.\n"
+      puts "Sorry that is not a valid move. Please select an integer between 1 and 9.\n"
```
Comment (jchung722) [inline ticTacToe.rb]: Nice! I like this change!

Comment (evan-m-jackson) [reply ticTacToe.rb]: Thanks! Sunny caught this for me - I have to do a better job proof reading!

File: lib/player_choice.rb (hunk 1 above)
Comment (evan-m-jackson) [reply lib/player_choice.rb]: Yes, I like that much better!

File: spec/board_spec.rb (hunk 2 above)
Comment (evan-m-jackson) [reply spec/board_spec.rb]: Agreed.

File: spec/game_over_spec.rb (hunk 3 above)
Comment (evan-m-jackson) [reply spec/game_over_spec.rb:62]: Agreed. Seems a lot more organized.

File: spec/ticTacToe_spec.rb (hunk 5 above)
Comment (evan-m-jackson) [reply spec/ticTacToe_spec.rb]: Sure did. I'll make sure all of the class specs are in the appropriate files.

File: spec/player_choice_spec.rb (hunk 4 above)
Comment (evan-m-jackson) [reply spec/player_choice_spec.rb]: Agreed. Will try to make test names more specific.

**Review by jchung722 — APPROVED** [review]
Hey @evan-m-jackson! I'm giving an approval because I think this is a good point to merge before making any new changes. One ask before doing so is I noticed that there is an inconsistency with indents in a lot of these files - could you make sure the indents for classes/methods/tests align to make this more readable? You could also use a linter to get the job done

File: spec/game_over_spec.rb [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+      @board = %w[X X X O O 6 7 8 9]
+      expect { @game_over.game_over_message(@board) }.to output("GAME OVER! Player X has won!!\n").to_stdout
+    end
+
+    it 'returns a message that says Player O has won' do
+      @board = %w[X X 3 O O O X 8 9]
+      expect { @game_over.game_over_message(@board) }.to output("GAME OVER! Player O has won!!\n").to_stdout
+    end
+
+    it 'returns a message that says Player O has won' do
```
Comment (jchung722) [inline spec/game_over_spec.rb]: I think this should be "message that says game is a draw"

File: spec/play_spec.rb [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        input = StringIO.new("a")
+        @player = PlayerChoice.new(input: input)
+        @play = Play.new(@board, @cpu, @player, @end_of_game)
+        expect(@play.player_picks_a_spot).to be false 
+    end
+ end
+
+ describe '.while_loop_for_player_picking_spot' do
+    it 'Method continues to run until a valid option is chosen' do
+        @play.while_loop_for_player_picking_spot
```
Comment (jchung722) [inline spec/play_spec.rb]: I think this test and the following need assertions to validate that things are working as described?

File: spec/game_over_spec.rb (hunk 7 above)
Comment (evan-m-jackson) [reply spec/game_over_spec.rb]: Agreed.
